reward/stakingInfo_test.go

```diff
@@ -0,0 +1,185 @@
+	"math"
+	"testing"
+)
+
+func TestStakingInfo_GetIndexByNodeId(t *testing.T) {
+	testdata := []common.Address{
+		common.StringToAddress("0xB55e5986b972Be438b4A91d6e8726aA50AD55EDc"),
+		common.StringToAddress("0xaDfc427080B4a66b5a629cd633d48C5d734572cA"),
+		common.StringToAddress("0x994daB8EB6f3FaE044cC0c9a0AB1A038e136b0B6"),
+		common.StringToAddress("0xD527822212Fded72c5fE89f46281d5355BD58235"),
+	}
+	testCases := []struct {
+		address common.Address
+		index   int
+	}{
+		{common.StringToAddress("0xB55e5986b972Be438b4A91d6e8726aA50AD55EDc"), 0},
+		{common.StringToAddress("0xaDfc427080B4a66b5a629cd633d48C5d734572cA"), 1},
+		{common.StringToAddress("0x994daB8EB6f3FaE044cC0c9a0AB1A038e136b0B6"), 2},
+		{common.StringToAddress("0xD527822212Fded72c5fE89f46281d5355BD58235"), 3},
+		{common.StringToAddress("0x027AbB8c9f952cfFf01B1707fF14E2CB5D439502"), AddrNotFoundInCouncilNodes},
+	}
+
+	stakingInfo, error := newEmptyStakingInfo(0)
+	if error != nil {
+		assert.FailNow(t, error.Error())
+	}
+	stakingInfo.CouncilNodeAddrs = testdata
+
+	for i := 0; i < len(testCases); i++ {
+		result := stakingInfo.GetIndexByNodeId(testCases[i].address)
+		assert.Equal(t, testCases[i].index, result)
+	}
+}
+
+func TestStakingInfo_GetStakingAmountByNodeId(t *testing.T) {
+	testdata := struct {
+		address       []common.Address
+		stakingAmount []uint64
+	}{
+		[]common.Address{
+			common.StringToAddress("0xB55e5986b972Be438b4A91d6e8726aA50AD55EDc"),
+			common.StringToAddress("0xaDfc427080B4a66b5a629cd633d48C5d734572cA"),
+			common.StringToAddress("0x994daB8EB6f3FaE044cC0c9a0AB1A038e136b0B6"),
+			common.StringToAddress("0xD527822212Fded72c5fE89f46281d5355BD58235"),
+		},
+		[]uint64{
+			100, 200, 300, 400,
+		},
+	}
+	testCases := []struct {
+		address       common.Address
+		stakingAmount uint64
+	}{
+		{common.StringToAddress("0xB55e5986b972Be438b4A91d6e8726aA50AD55EDc"), 100},
+		{common.StringToAddress("0xaDfc427080B4a66b5a629cd633d48C5d734572cA"), 200},
+		{common.StringToAddress("0x994daB8EB6f3FaE044cC0c9a0AB1A038e136b0B6"), 300},
+		{common.StringToAddress("0xD527822212Fded72c5fE89f46281d5355BD58235"), 400},
+		{common.StringToAddress("0x027AbB8c9f952cfFf01B1707fF14E2CB5D439502"), 0},
+	}
+
+	stakingInfo, error := newEmptyStakingInfo(0)
+	if error != nil {
+		assert.FailNow(t, error.Error())
+	}
+	stakingInfo.CouncilNodeAddrs = testdata.address
+	stakingInfo.CouncilStakingAmounts = testdata.stakingAmount
+
+	for i := 0; i < len(testCases); i++ {
+		result := stakingInfo.GetStakingAmountByNodeId(testCases[i].address)
+		assert.Equal(t, testCases[i].stakingAmount, result)
+	}
+}
+
+func TestCalcGiniCoefficient(t *testing.T) {
+	testCase := []struct {
+		testdata []uint64
+		result   float64
+	}{
+		{[]uint64{1, 1, 1}, 0.0},
+		{[]uint64{0, 8, 0, 0, 0}, 0.8},
+		{[]uint64{5, 4, 3, 2, 1}, 0.27},
+	}
+
+	for i := 0; i < len(testCase); i++ {
+		result := CalcGiniCoefficient(testCase[i].testdata)
+
+		if result != testCase[i].result {
```

Comment:
We can simplify this with `assert.NotEqual`.

Reply:
reflected it.